streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroStreamsSnapshotDeserializer.java

```diff
@@ -65,7 +65,10 @@ private Object convertValue(Object deserializedObj) {
             List<Schema.Field> fields = indexedRecord.getSchema().getFields();
             ImmutableMap.Builder<String, Object> keyValues = ImmutableMap.builder();
             for (Schema.Field field : fields) {
-                keyValues.put(field.name(), convertValue(indexedRecord.get(field.pos())));
+                Object currentValue = convertValue(indexedRecord.get(field.pos()));
+                if (currentValue != null) {
```

Comment:
@priyank5485 , if you are changing the implementation you also need to make sure to re-implement all the methods that could potentially modify the map (like the entrySet etc). Also the Immutablemap.CopyOf does a reference copy if the map is already immutable, and these kind of optimization is really needed in our transformations where we always emit a new event. Allowing null values opens up possiblility of NPE so thats one of the reasons why the guava immutable map disallows it. Another option would be to insert something like `Optional.empty()` instead of null value and have it well defined in our contract/javadocs. The receivers can check for `value == Optional.empty()` than `value == null` wherever they want null semantics.